presto-jdbc/src/test/java/io/prestosql/jdbc/TestJdbcConnection.java

```diff
@@ -289,6 +289,29 @@ public void testApplicationName()
         }
     }
 
+    @Test
+    public void testSource()
+            throws SQLException
+    {
+        try (Connection connection = createConnection("source=testing")) {
+            assertConnectionSource(connection, "testing");
+        }
+
+        try (Connection connection = createConnection("source=testing&applicationNamePrefix=fruit:")) {
+            assertConnectionSource(connection, "testing");
+        }
+
+        try (Connection connection = createConnection("source=testing")) {
+            connection.setClientInfo("ApplicationName", "testingApplicationName");
+            assertConnectionSource(connection, "testing");
+        }
+
+        try (Connection connection = createConnection("source=testing&applicationNamePrefix=fruit:")) {
+            connection.setClientInfo("ApplicationName", "testingApplicationName");
+            assertConnectionSource(connection, "testing");
+        }
```

Comment:
can you please also add tests for `assertConnectionSource(connection, "testing");` where `source` parameter is not given in connection string.

Reply:
yes, done.